mlflow/tracing/fluent.py

```diff
@@ -275,7 +279,19 @@ def get_trace(request_id: str) -> Optional[Trace]:
     Returns:
         A :py:class:`mlflow.entities.Trace` objects with the given request ID.
     """
-    return TRACE_BUFFER.get(request_id, None)
+    # Try to get the trace from the in-memory buffer first
+    if trace := TRACE_BUFFER.get(request_id, None):
+        return trace
+
+    try:
+        return MlflowClient().get_trace(request_id, display=False)
```

Comment:
One question is whether or not we should display trace UI. Currently, `mlflow.get_trace()` does not display, while `mlflow.MlflowClient().get_trace()` does, and this PR keeps the same behavior.

Reply:
Do we need to consider any mismatch cases? e.g. trace exists in TRACE_BUFFER but not in backend or vice versa

Reply:
Oh the opposite direction is just what you implement in this PR

Reply:
>  What about trace exists in TRACE_BUFFER but not in backend? It seems not likely to happen

Yea I agree, it's much less likely than the opposite. Also it's probably ok to return the in-memory trace in that case.